### DISCUSSION TITLE
I'm trying to add a custom layer with filter

### DISCUSSION THREAD

**Discussion by tayo-delivus** [placemark/placemark#84] [Q&A]
I'm trying to add a custom layer. 
You want to create a custom point and add a point that contains an icon image when clicked.
We've already added the icon-image.
properties contains {“star”:true} in the properties.

When I apply the code below, I get the error “filter[3][1]: string expected, array found” when I apply it. 
How can I fix it? 

```
// custom layer
{
    id: FEATURES_CUSTOM_POINT_LAYER_NAME,
    type: "symbol",
    source: FEATURES_SOURCE_NAME,
    layout: {
      "icon-image": "star", 
      "icon-size": 1, 
    },
    filter: [
      "all",
      ["==", "$type", "Point"],
      ["has", "star"],
      ["==", ["get", "star"], "true"],
    ],
},
```

- `[“has”, “entrance”]` doesn't seem to apply either.
- When applying only `[“==”, “$type”, “Point”]`, the icon looks fine.
- `{“star”:“true”}` in properties also doesn't work also doesn't work.

**Comment by tmcw** [comment]
The correct line would be

```
      ["==", "star", "true"],
```

The error message says this - `filter[3][1]: string expected, array found`: look at the 3rd (zero-indexed) item in the filter, and the 1st (zero-indexed) item in that, and it's the `["get", "star"]`. See an [example of doing this filtering](https://docs.mapbox.com/mapbox-gl-js/example/filter-markers/) and it uses the property name as the second item. The GL style spec documentation is definitely hard to follow, though.

I can't really lend any advice for the other filters, that's specific to the data and like this, is more about following the Mapbox GL Style Spec rather than anything specific to Placemark.
